In [1]:
import numpy as np
import pandas as pd
import sklearn as sk
import matplotlib.pyplot as plt
from sklearn import linear_model
from sklearn.model_selection import train_test_split
from sklearn import metrics

In [2]:
df = pd.read_csv("mlfolder/CSV_Files/DiabetesData_PimaIndians - Copy.csv")

In [3]:
X = np.array(df.drop("Diabetes",axis=1)) 
y =  df["Diabetes"]  
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=.2, random_state=42)

In [4]:
%config Completer.use_jedi = False  ## for mac

In [5]:
np.sum(y=='neg')

np.int64(475)

In [6]:
np.sum(y=='pos')

np.int64(249)

In [7]:
from sklearn.svm import SVC

In [8]:
from sklearn.model_selection import GridSearchCV

In [9]:
SVCModel=SVC(gamma='scale')

In [10]:
param_grid = {
    'kernel':['linear', 'rbf', 'poly'],
    'degree':[2, 3, 4],
    'C': [1, 10, 50, 100, 1000]
}

In [11]:
grid_search = GridSearchCV(
    SVCModel, param_grid, cv=5, scoring='accuracy')

In [12]:
#grid_search = GridSearchCV(SVCModel, param_grid, cv=5, scoring='f1')

In [13]:
%%time

UsageError: %%time is a cell magic, but the cell body is empty. Did you mean the line magic %time (single %)?


In [14]:
%%time
grid_search.fit(X_train, y_train)

CPU times: user 3min 30s, sys: 1.04 s, total: 3min 31s
Wall time: 3min 32s


GridSearchCV(cv=5, estimator=SVC(),
             param_grid={'C': [1, 10, 50, 100, 1000], 'degree': [2, 3, 4],
                         'kernel': ['linear', 'rbf', 'poly']},
             scoring='accuracy')

In [15]:
bestModel=grid_search.best_estimator_

In [16]:
grid_search.best_score_

np.float64(0.7686206896551724)

In [47]:
grid_search.best_params_

{'C': 50, 'degree': 3, 'kernel': 'poly'}

In [48]:
grid_search.best_estimator_.C

50

In [17]:
import xgboost as xgb

XGBoostError: 
XGBoost Library (libxgboost.dylib) could not be loaded.
Likely causes:
  * OpenMP runtime is not installed
    - vcomp140.dll or libgomp-1.dll for Windows
    - libomp.dylib for Mac OSX
    - libgomp.so for Linux and other UNIX-like OSes
    Mac OSX users: Run `brew install libomp` to install OpenMP runtime.

  * You are running 32-bit Python on a 64-bit OS

Error message(s): ['dlopen(/Users/easwaran/demo1/env/lib/python3.6/site-packages/xgboost/lib/libxgboost.dylib, 6): Library not loaded: /usr/local/opt/libomp/lib/libomp.dylib\n  Referenced from: /Users/easwaran/demo1/env/lib/python3.6/site-packages/xgboost/lib/libxgboost.dylib\n  Reason: image not found']


In [14]:
%%time

from sklearn.experimental import enable_halving_search_cv
from sklearn.model_selection import HalvingGridSearchCV

halving_cv = HalvingGridSearchCV(
    xgb_cl, param_grid, scoring="roc_auc", n_jobs=-1, min_resources="exhaust", factor=3
)

_ = halving_cv.fit(X, y)

NameError: name 'xgb_cl' is not defined

In [49]:
grid_search.best_estimator_.kernel                                

'poly'

In [50]:
metrics.accuracy_score(y_test, bestModel.predict(X_test))

0.8068965517241379

In [51]:
df

Glucose  DiastolicPressue   BMI  Age Diabetes
0        148                72  33.6   50      pos
1         85                66  26.6   31      neg
2        183                64  23.3   32      pos
3         89                66  28.1   21      neg
4        137                40  43.1   33      pos
..       ...               ...   ...  ...      ...
719      101                76  32.9   63      neg
720      122                70  36.8   27      neg
721      121                72  26.2   30      neg
722      126                60  30.1   47      pos
723       93                70  30.4   23      neg

[724 rows x 5 columns]

In [52]:
grid_search.best_estimator_.decision_function

<bound method BaseSVC.decision_function of SVC(C=50, kernel='poly')>

### Best accuraacy, 80.7% is achieved by a cubic polynomial kernel with C=50.

In [53]:
from sklearn.linear_model import LogisticRegression

In [54]:
lr = LogisticRegression()

In [55]:
lr.fit(X_train, y_train)

LogisticRegression()

In [56]:
metrics.accuracy_score(y_test, lr.predict(X_test))

0.8068965517241379

In [57]:
print(metrics.classification_report(y_test, lr.predict(X_test)))

              precision    recall  f1-score   support

         neg       0.84      0.89      0.87       102
         pos       0.70      0.60      0.65        43

    accuracy                           0.81       145
   macro avg       0.77      0.75      0.76       145
weighted avg       0.80      0.81      0.80       145



In [58]:
print(metrics.classification_report(y_test, bestModel.predict(X_test)))

              precision    recall  f1-score   support

         neg       0.84      0.89      0.87       102
         pos       0.70      0.60      0.65        43

    accuracy                           0.81       145
   macro avg       0.77      0.75      0.76       145
weighted avg       0.80      0.81      0.80       145

